stable_baselines/bench/monitor.py

```diff
@@ -14,7 +14,7 @@ class Monitor(Wrapper):
     EXT = "monitor.csv"
     file_handler = None
 
-    def __init__(self, env, filename, allow_early_resets=False, reset_keywords=(), info_keywords=()):
+    def __init__(self, env, filename, allow_early_resets=True, reset_keywords=(), info_keywords=()):
```

Comment:
Come think of it, what is the purpose of `allow_early_resets`? Compatibility with some other wrappers?

Reply:
>what is the purpose of allow_early_resets?

I'm not sure but this happens a lot when using VecEnv.

>Sounds like a thing for the common-refactoring.

I totally agree on that ;)

Reply:
>I'm not sure but this happens a lot when using VecEnv.

As in why do we have a flag for allowing early resets? I do not see why it should not be always allowed. If environment prevents it for whatever reason, it should throw the exceptions and likes.

Reply:
>As in why do we have a flag for allowing early resets?

legacy code.... again